reproject/utils.py

```diff
@@ -142,9 +142,11 @@ def parse_output_projection(output_projection, shape_in=None, shape_out=None, ou
                     "Need to specify shape since output header "
                     "does not contain complete shape information"
                 )
-    elif isinstance(output_projection, BaseHighLevelWCS):
+    elif isinstance(output_projection, (BaseLowLevelWCS, BaseHighLevelWCS)):
         wcs_out = output_projection
-        if shape_out is None:
+        if getattr(wcs_out, "array_shape") is not None:
+            shape_out = wcs_out.array_shape
```

Comment:
The issue is that not all `HighLevelWCS` classes also implement the low level API (notably the wrapper doesn't). So I think you might not be able to get to `array_shape` if the `output_projection` is an instance of wrapper. (You would have to go through `low_level_wcs`.

It might be worth standardising on low level?

Reply:
We've been allowing high level WCS for a little while so I think we just have to support both (in fact we rely on the high level API internally). I've expanded the test suite and made sure things work for both now.